wsmaster/che-core-api-workspace/src/main/java/org/eclipse/che/api/workspace/server/wsnext/WorkspaceNextObjectsRetriever.java

```diff
@@ -192,79 +197,79 @@ private CheFeature getFeature(String featureName, String featureVersion)
 
     if (!parameters.isEmpty()) {
       throw new InfrastructureException(
-          "Parameters not supported by services found: " + parameters.keySet());
+          "Parameters not supported by che plugins found: " + parameters.keySet());
     }
 
-    return services;
+    return plugins;
   }
 
-  private Collection<CheService> getServices(Collection<CheFeature> features)
+  private Collection<ChePlugin> getPlugins(Collection<CheFeature> features)
       throws InfrastructureException {
-    Map<String, CheService> services = new HashMap<>();
+    Map<String, ChePlugin> plugins = new HashMap<>();
     for (CheFeature feature : features) {
-      for (CheServiceReference serviceReference : feature.getSpec().getServices()) {
-        String serviceName = serviceReference.getName();
-        String serviceVersion = serviceReference.getVersion();
-        String key = serviceName + '/' + serviceVersion;
-        if (!services.containsKey(key)) {
-          CheService service = getService(serviceName, serviceVersion);
-          services.put(key, service);
+      for (ChePluginReference pluginReference : feature.getSpec().getServices()) {
+        String pluginName = pluginReference.getName();
+        String pluginVersion = pluginReference.getVersion();
+        String key = pluginName + '/' + pluginVersion;
+        if (!plugins.containsKey(key)) {
+          ChePlugin plugin = getPlugin(pluginName, pluginVersion);
+          plugins.put(key, plugin);
         }
       }
     }
-    return services.values();
+    return plugins.values();
   }
 
   private boolean isParameter(EnvVar envVar) {
     return PARAMETER_ENV_VAR_VALUE.matcher(envVar.getValue()).matches();
   }
 
   /**
-   * Returns map where key is concatenation of service name, version and parameter name separated by
-   * slash symbol. < serviceName/serviceVersion/${parameter} > to < parameterValue >
+   * Returns map where key is concatenation of plugin name, version and parameter name separated by
+   * slash symbol. < pluginName/pluginVersion/${parameter} > to < parameterValue >
    */
-  private Map<String, List<String>> getServicesParameters(Collection<CheFeature> features) {
+  private Map<String, List<String>> getPluginsParameters(Collection<CheFeature> features) {
     Map<String, List<String>> parameters = new HashMap<>();
     for (CheFeature feature : features) {
-      for (CheServiceReference serviceReference : feature.getSpec().getServices()) {
-        if (serviceReference.getParameters().isEmpty()) {
+      for (ChePluginReference pluginReference : feature.getSpec().getServices()) {
```

Comment:
Should be `feature.getSpec().getServices()` method renamed to feature.getSpec().getFeature()?

Reply:
I'm not going to update Feature related model as we are going to remove this entity from the model according to the discussion in workspace next walking skeleton https://github.com/eclipse/che/issues/10123